builtin/string.go

```diff
@@ -346,12 +348,20 @@ var _ = method(string_NumberQ, "()")
 
 func string_NumberQ(this Value) Value {
 	// see also Lexer.number
-	return SuBool(numberPat.Matches(ToStr(this)))
+	lexer := lexer.NewLexer(ToStr(this))
+	item := lexer.Next()
+	if item.Token == tok.Add || item.Token == tok.Sub {
+		item = lexer.Next()
+	}
+	if item.Token == tok.Number {
+		item = lexer.Next()
+		if item.Token == tok.Eof {
+			return True
+		}
+	}
+	return False
```

Comment:
It's minor, but I would avoid the extra nesting and keep this more "straight line".

```
	if item.Token != tok.Number {
		return False
	}
	item = lexer.Next()
	return item.Token == tok.Eof
```

Reply:
The function returns a value and the `==` would evaluate to a bool, should I just wrap the return value in a SuBool?